shared-module/displayio/__init__.c

```diff
@@ -245,21 +245,6 @@ void displayio_gc_collect(void) {
     }
 }
 
-void displayio_area_expand(displayio_area_t *original, const displayio_area_t *addition) {
```

Comment:
Did you consider or measure the effective impact of switching from 4 conditioned writes to multiple conditions followed by 4 branching unconditional writes?
Probably in this case it is okay because it is per-frame, but seemingly innocuous change can have dramatic pixel fill rate or frame latency impact in this area so I’m somewhat skeptical of doing more work than is necessary.

Reply:
That's a great point.  My main motivation was to fix an issue where when area1 was combined with an empty area (defined as empty by x1==x2==0) that it would return an area with x1=0 and x2=area1.x2.  This caused unnecessary dirty rectangle redrawing.

But your point is about performance of the `union` command vs. the code in `expand`.  I didn't look at any performance differences.  I've never done any performance testing down at the C-level.  If you give a good example of how best to do this to make the comparison, and I'll check it out.

Reply:
I think a test for this would be a chore. Native time functions have a 32khz resolution iirc and either call should surely take less than 31 microseconds? You could time a frame with several 1px overlapping rectangles but there’s quite a bit of noise in that aggregate signal. I’ve wrapped code up higher in the displayio draw stack to A/B test things in this way but I do not have a clear example for you to follow.
Last time I was actively adding to vectorio I wanted to make a more convenient way to use a high precision hardware timer for things like this.
I really don’t think there’s a problem with this code, I just appreciate you taking an active interest and hope you also always defend (and improve!) the pixel fill rate :-)